ingestion/src/metadata/ingestion/source/dashboard/quicksight/metadata.py

```diff
@@ -209,124 +229,353 @@ def yield_dashboard_chart(
+                    )
+
+    def _get_datamodel(self, datamodel_id: str):
+        datamodel_fqn = fqn.build(
+            self.metadata,
+            entity_type=DashboardDataModel,
+            service_name=self.context.get().dashboard_service,
+            data_model_name=datamodel_id,
+        )
+        if datamodel_fqn:
+            return self.metadata.get_by_name(
+                entity=DashboardDataModel,
+                fqn=datamodel_fqn,
+            )
+        return None
+
+    def yield_dashboard_lineage_details(  # pylint: disable=too-many-locals
+        self, dashboard_details: DashboardDetail, db_service_name: str
+    ) -> Iterable[Either[AddLineageRequest]]:
+        """
+        Get lineage between dashboard and data sources
+        """
+        db_service_entity = self.metadata.get_by_name(
+            entity=DatabaseService, fqn=db_service_name
+        )
+        for datamodel in self.data_models:
+            try:
+                data_model_entity = self._get_datamodel(
+                    datamodel_id=datamodel.DataSource.DataSourceId
+                )
+                if isinstance(
+                    datamodel.DataSource.data_source_resp, DataSourceRespQuery
+                ):
+                    yield from self._yield_lineage_from_query(
+                        data_model_entity,
+                        datamodel.DataSource,
+                        dashboard_details,
+                        db_service_entity,
+                    )
+                elif isinstance(
+                    datamodel.DataSource.data_source_resp, DataSourceRespS3
+                ):
+                    yield from self._yield_lineage_from_s3(
+                        data_model_entity, datamodel.DataSource, dashboard_details
+                    )
+                elif isinstance(datamodel.DataSource.data_source_resp, DataSourceResp):
+                    yield from self._yield_lineage_from_table(
+                        data_model_entity,
+                        datamodel.DataSource,
+                        dashboard_details,
+                        db_service_entity,
+                    )
+            except Exception as exc:  # pylint: disable=broad-except
+                yield Either(
+                    left=StackTraceError(
+                        name="Lineage",
+                        error=f"Error to yield dashboard lineage details for DB service name [{db_service_name}] and dashboard_name [{dashboard_details.Name}]: {exc}",
+                        stackTrace=traceback.format_exc(),
+                    )
+                )
+
+    def _get_column_info(self, data_model: DescribeDataSourceResponse):
+        """"""
+        datasource_columns = []
+        for field in data_model.DataSource.data_source_resp.columns or []:
+            try:
+                parsed_fields = {
+                    "name": field.get("Name"),
+                    "dataType": field.get("Type"),
```

Comment:
lets use something similar to this to parse the type otherwise we may get the pydantic errors for types which are supported by quicksight but now us, this also eliminates need of adding a new type to table.json 

https://github.com/open-metadata/OpenMetadata/blob/ef0b03d4a90d48bd1d64be84e91d212eaa99f53b/ingestion/src/metadata/ingestion/source/dashboard/superset/mixin.py#L244